### DISCUSSION TITLE
Snapshot Prechange null in Webhook generated by Custom Script Change

### DISCUSSION THREAD

**Discussion by boomsfib** [netbox-community/netbox#10397] [Q&A] [answered]
Netbox v3.1.11

I've written a custom script in NetBox to make changes to a dcim interface.  I've also got a webhook for updates to object type dcim interface that initiates an Ansible playbook to make the config changes for the interface on the physical device.

Issue is, when using the script to make the interface updates in NetBox, the webhook's payload shows prechange for snapshots as being null and only postchange is defined.  If I make the same updates to the interface via the GUI, prechange is defined.

Is this to be expected or could this be a bug?

**Comment by kkthxbye-code** [comment, marked as answer]
You have to call `object.snapshot()` before modifying it. Netbox does this internally when fetching the object, but you must do it yourself in custom scripts.

There's an accepted issue for documenting this in the custom script documentation.

https://github.com/netbox-community/netbox/issues/9651

**Reply by boomsfib** [reply]
Thank you so much for the reference.  That solved my issue.
